lib/puppet/util/yaml.rb

```diff
@@ -24,7 +24,11 @@ class YamlLoadError < Puppet::Error; end
   # @raise [YamlLoadException] If deserialization fails.
   # @return The parsed YAML, which can be Hash, Array or scalar types.
   def self.safe_load(yaml, allowed_classes = [], filename = nil)
-    data = YAML.safe_load(yaml, allowed_classes, [], true, filename)
+    if Psych::VERSION >= "3.1.0"
```

Comment:
Thanks for the contribution @tambry!

Comparing strings can lead to surprises:

```
irb(main):015:0> "3.1" >= "3.1.0"
=> false
```

Might be better off using `to_f`, which we use in other places like `RUBY_VERSION.to_f`:

```ruby
if Psych::VERSION.to_f >= 3.1
```

The psych dependency is getting pulled in due to `rdoc`:

```
    rdoc (6.4.0)
      psych (>= 4.0.0)
```

And apparently `rdoc` doesn't follow SemVer: https://github.com/ruby/rdoc/issues/857#issuecomment-1003787128

Could you update https://github.com/puppetlabs/puppet/blob/6.x/Gemfile#L32 so it excludes 6.0.4, something like:

```
gem 'rdoc', ['~> 6.0', '< 6.0.4'], require: false, platforms: [:ruby]
```

If you'd like attribution in our release notes, please file a PUP ticket and update the commit summary and GH PR title with `(PUP-XXX) Replace  deprecated ...` Thanks!

Reply:
Thanks for the review! I've made all the suggested changes.